### PR TITLE
Store the full translation resource for translated resources

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Updates the Redux translations store to store the full translations resource for project, workflow, field guide, mini course, tutorial and project pages. Previously only the translation strings were stored. The store structure is now `state.strings[translated_type][translated_id]`. Each type can now store translations for multiple resources, where previously only one resource translation could be stored at a time.
- Updates the project, classifier and task components to read translations by translated resource ID. This should make it impossible to accidentally confuse workflow translations, for example.
- Updates tests to use the new store structure.

Staging branch URL: https://pr-5137.pfe-preview.zooniverse.org

Snapshot Hoge Veluwe is a project that uses translations: https://pr-5137.pfe-preview.zooniverse.org/projects/y-dot-liefting/snapshot-hoge-veluwe?language=nl&env=production

TODO:

- [x] project page translations
- [x] consolidate the different translation components into one generic component.
- [x] consolidate `translations.load` (takes a single ID as an argument) and `translations.loadTranslations` (takes an array of IDs) into a single action for loading translations.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by mcbouslog** [comment]
This is looking really good to me! No issues with a little testing. Makes more sense to me, though not sure if just more familiar with related code and translations structure or slow on understanding initially.

**Comment by mcbouslog** [comment]
Once I started adding 'resource' in my head after 'translated' for many properties (i.e. `translated_type`, `translated_id`) things became more clear for me. Just noting if anyone new to translations comes across this comment.

**Comment by eatyourgreens** [comment]
Should we add the workflow translation `translated_id` to the classification, if only for quick validation because it should always match `classification.links.workflow`?

**Comment by eatyourgreens** [comment]
I think this is ready now.

File: app/redux/ducks/translations.js [hunk 1]
```diff
@@ -107,50 +99,7 @@ export function listLanguages(translated_type, translated_id) {
   };
 }
 
-export function load(resource_type, translated_id, language) {
-  counterpart.setLocale(language);
-  const translated_type = resource_type === 'minicourse' ? 'tutorial' : resource_type;
```
Comment (eatyourgreens) [inline app/redux/ducks/translations.js]: This line shouldn't have been deleted.

**Pull request merged**